### DISCUSSION TITLE
Budibase apps lost in Docker Azure Container

### DISCUSSION THREAD

**Discussion by RicardoDibo** [Budibase/budibase#9167] [Help]
We recently deployed budibase inside an Azure App Service B1 as in the budibase documentation, but the App Service machine was restarted and we lost all our created aps, we did some tests and every time it restarts the budibase App Services, everything that was created are deleted, do you know how I can solve this?

**Comment by shogunpurple** [comment]
Hi @RicardoDibo - did you follow the persistent storage guide here? https://docs.budibase.com/docs/azure-container-instances

**Reply by RicardoDibo** [reply]
Hello, we are only using the App Service system, and now we are configuring this Container. How can I just use the AppService with data persistence without configuring the Container? (We used to read and configure as per this link "https://docs.budibase.com/docs/azure-app-service#docker-settings").
Another point is do I need only the Container or both systems to keep the apps, data, and users? (Appservice and Container Instances).

**Reply by shogunpurple** [reply]
Azure app service does not have any built in persistent storage. A lot of apps don't really need this, because you can just use azure blob storage combined with an external database. However budibase needs persistent storage for several functions including its database.

You don't need both - it's just one or the other. Apart from the guide we have there for configuring ACI with persistent storage, I'm not sure if AAS with persistent storage is a configuration budibase supports. If you must use AAS, then this guide here may be of use - seems like you can possibly mount the file path in the container to azure storage.

https://learn.microsoft.com/en-us/azure/app-service/configure-connect-to-azure-storage

**Reply by RicardoDibo** [reply]
Perfect, thank you very much for your help, with that we had a light at the end of the tunnel, let's configure it with direct ACI without AAS. I am very grateful for the help. If in doubt, I'll get back to you here. Happy Holidays.
